File: frontend/src/types/schemas.ts
// Mirrors backend/app/schemas.py. Keep in sync — this is the cross-team contract.

export interface RecurringExpense {
  name: string;
  monthly_amount: number;
  category: string;
}

export interface PlanRequest {
  country: string;              // ISO-3166-1 alpha-2
  currency: string;
  monthly_income: number;
  recurring: RecurringExpense[];
  savings_goal: number;
  deadline: string;             // ISO date (YYYY-MM-DD)
  current_savings: number;
}

export interface CategoryBudget {
  category: string;
  monthly_amount: number;
  is_essential: boolean;
}

export interface PlanResponse {
  id?: number;
  daily_budget: number;
  weekly_budget: number;
  monthly_budget: number;
  months_to_goal: number;
  projected_date: string;
  on_track: boolean;
  breakdown: CategoryBudget[];
  total_essentials: number;
  total_recurring: number;
  monthly_savings: number;
}

export interface Country {
  code: string;
  name: string;
  currency: string;
}

File: backend/app/schemas.py
"""Shared data contract. Keep in sync with frontend/src/types/schemas.ts."""
from datetime import date

from pydantic import BaseModel, Field


class RecurringExpense(BaseModel):
    name: str
    monthly_amount: float = Field(ge=0)
    category: str  # "subscription" | "fitness" | "transport" | "other"


class PlanRequest(BaseModel):
    country: str                                         # ISO-3166-1 alpha-2, e.g. "US"
    currency: str                                        # e.g. "USD"
    monthly_income: float = Field(gt=0)
    recurring: list[RecurringExpense] = Field(default_factory=list)
    savings_goal: float = Field(gt=0)
    deadline: date
    current_savings: float = 0.0


class CategoryBudget(BaseModel):
    category: str                                        # housing | food | utilities | transport | recurring | discretionary | savings
    monthly_amount: float
    is_essential: bool


class PlanResponse(BaseModel):
    id: int | None = None
    daily_budget: float                                  # recommended discretionary spend per day
    weekly_budget: float
    monthly_budget: float
    months_to_goal: float
    projected_date: date
    on_track: bool                                       # projected_date <= deadline
    breakdown: list[CategoryBudget]
    total_essentials: float
    total_recurring: float
    monthly_savings: float
